recipes/gatk4/build_main.sh

```diff
@@ -1,15 +1,37 @@
-#!/bin/bash
 
-BINARY_HOME=$PREFIX/bin
+#!/bin/bash
+JAVA_VERSION=17.0.9_9
 PACKAGE_HOME=$PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM
+JAVA_DIR=$PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/java-for-gatk
 
 mkdir -p $PREFIX/bin
 mkdir -p $PACKAGE_HOME
+mkdir -p $JAVA_DIR
+
+#
+# We have downloaded Java from Adoptium as none of the OpenJDK-17 builds in Conda-forge are compatible with GATK. 
+# To prevent conflicts with existing Java installations, we have adjusted the GATK wrapper script to utilize the downloaded 
+# Java executable directly without needing to add it to the system path.
+#
```

Comment:
Wow, this is interesting. Just for my education why is this?

Reply:
When I tried to used openjdk-17 builds on conda-forge (that were also compatible with other packages in the environment), none of them were compatble with gatk itself. I got this error message:
`org/broadinstitute/hellbender/Main has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 52.0`